### PR TITLE
AYS-130 Invalidated Refresh Token Flow Fix

### PR DESCRIPTION
**Pull request by moaydogdu**
Please provide a brief description of the changes made in this pull request.

### Checklist

Before submitting your pull request, ensure the following:

- [X] **Title and Branch Naming Conventions:**
    - The pull request title follows the
      standard: [Pull Request Naming Conventions](CONTRIBUTING.md#pull-request-naming-conventions).
    - The branch name follows one of the conventions outlined in
      the [Branch Naming Conventions](CONTRIBUTING.md#branch-naming-conventions).

- [X] **Local Testing:**
    - I have tested my changes locally on Postman, and they are working as expected.

- [X] **Code Quality:**
    - The code is formatted according to the project's coding guidelines and style.
    - The code has been reviewed to ensure its quality.
    - The code does not contain any issues flagged by SonarLint.

- [X] **Documentation:**
    - Necessary documentation has been added or existing documentation has been updated, specifically detailing changes
      made in Postman.

- [X] **Testing:**
    - Relevant unit tests have been written and included.
    - Relevant integration tests have been written and included.

- [X] **Reviewers and Assignees:**
    - Default reviewers have been assigned to this pull request.
    - Assignees have been added if necessary.

- [X] **Labels and Associations:**
    - No specific actions are required in the Labels and Associations section for this pull request.

- [X] **Jira Issue Transition:**
    - If this pull request is related to a Jira issue, ensure that the issue is moved to the "In Review" status on the
      Jira board.

### CODE REVIEW

File: src/test/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImplTest.java [hunk 1]
```diff
@@ -256,8 +263,68 @@ void givenInvalidRefreshToken_whenTokenNotVerifiedOrValidate_thenThrowTokenNotVa
...
+                .findById(Mockito.anyString());
+        Mockito.verify(tokenService, Mockito.times(0))
+                .generate(Mockito.anyMap(), Mockito.anyString());
+
+    }
+
+    @Test
+    void givenInvalidRefreshToken_whenRefreshTokenAlreadyInvalidated_thenThrowTokenAlreadyInvalidatedException() {
+        // Given
+        String mockRefreshToken = """
```
Comment (ibrahimdenizz) [inline src/test/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImplTest.java:283]: @moaydogdu @agitrubard AysRandomTestUtil class'ına generateToken diye bir method ekliyebilir miyiz? Her defasında static bir değer kullanmak yerine generateToken methodun'dan alabiliriz.

File: src/test/java/com/ays/user/service/impl/UserAuthServiceImplTest.java [hunk 2]
```diff
@@ -232,8 +239,68 @@ void givenInvalidRefreshToken_whenTokenNotVerifiedOrValidate_thenThrowTokenNotVa
...
+        Mockito.verify(userRepository, Mockito.times(0))
+                .findById(Mockito.anyString());
+        Mockito.verify(tokenService, Mockito.times(0))
+                .generate(Mockito.anyMap(), Mockito.anyString());
+    }
+
+    @Test
+    void givenInvalidRefreshToken_whenRefreshTokenAlreadyInvalidated_thenThrowTokenAlreadyInvalidatedException() {
+        // Given
+        String mockRefreshToken = """
```
Comment (ibrahimdenizz) [inline src/test/java/com/ays/user/service/impl/UserAuthServiceImplTest.java:258]: @moaydogdu @agitrubard AysRandomTestUtil class'ına generateToken diye bir method ekliyebilir miyiz? Her defasında static bir değer kullanmak yerine generateToken methodun'dan alabiliriz.

**Review by ibrahimdenizz — CHANGES_REQUESTED** [review]

File: src/test/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImplTest.java (hunk 1 above)
Comment (agitrubard) [reply src/test/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImplTest.java:283]: Aslında burada bile isteye böyle bir şey kullanıyoruz, var olan token'ı bazen manuel olarak bozabiliyoruz fakat böyle bir şey yapılacaksa farklı bir issue'da bütün projeyi refactor etmek daha sağlıklı olur

File: src/test/java/com/ays/user/service/impl/UserAuthServiceImplTest.java (hunk 2 above)
Comment (agitrubard) [reply src/test/java/com/ays/user/service/impl/UserAuthServiceImplTest.java:258]: Aslında burada bile isteye böyle bir şey kullanıyoruz, var olan token'ı bazen manuel olarak bozabiliyoruz fakat böyle bir şey yapılacaksa farklı bir issue'da bütün projeyi refactor etmek daha sağlıklı olur

**Review by agitrubard — APPROVED** [review]

**Review by ibrahimdenizz — APPROVED** [review]

**Pull request merged**